plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbTypeMapping.java

```diff
@@ -876,6 +835,100 @@ public void testTimestampCoercion()
                 .execute(getQueryRunner(), trinoCreateAndInsert("test_timestamp_coercion"));
     }
 
+    @Test
+    public void testDatetime()
+    {
+        testDatetime(UTC);
+        testDatetime(jvmZone);
+        testDatetime(vilnius);
+        testDatetime(kathmandu);
+        testDatetime(TestingSession.DEFAULT_TIME_ZONE_KEY.getZoneId());
+    }
+
+    private void testDatetime(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(getTimeZoneKey(sessionZone.getId()))
+                .build();
+        createCommonDatetimeTimestampRoundTrip("datetime")
+                // before epoch
+                .addRoundTrip("datetime(3)", "TIMESTAMP '1958-01-01 13:18:03.123'", createTimestampType(3), "TIMESTAMP '1958-01-01 13:18:03.123'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1969-12-31 23:59:59.999995'", createTimestampType(6), "TIMESTAMP '1969-12-31 23:59:59.999995'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1969-12-31 23:59:59.999949'", createTimestampType(6), "TIMESTAMP '1969-12-31 23:59:59.999949'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1969-12-31 23:59:59.999994'", createTimestampType(6), "TIMESTAMP '1969-12-31 23:59:59.999994'")
+
+                //After epoch
+                .addRoundTrip("datetime(3)", "TIMESTAMP '3555-01-01 13:18:03.123'", createTimestampType(3), "TIMESTAMP '3555-01-01 13:18:03.123'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '5555-12-31 23:59:59.999995'", createTimestampType(6), "TIMESTAMP '5555-12-31 23:59:59.999995'")
```

Comment:
Remove as the roundtrips for after epoch already exist in a helper method.

Reply:
Helper method has roundtrips till max of epoch, this roundtrips exceeds max of epoch which is supported in `datetime` only. Updated the comments.